### PR TITLE
Fixing issue where interally closed socket wasn't 'closing' the recei…

### PR DESCRIPTION
**Pull request by berdon**
…ve loop. This would manifest in missed closed calls.

### CODE REVIEW

File: src/MorseL.Client.WebSockets/WebSocketClient.cs [hunk 1]
```diff
@@ -55,6 +55,8 @@ public WebSocketClient(string uri, Action<WebSocketClientOption> config=null, Ac
             };
             _internalWebSocket.Closed += (sender, args) =>
             {
+                // We're closing, kill the receive loop (which causes listeners to break out)
+                _internalCts.Cancel();
```
Comment (berdon) [inline src/MorseL.Client.WebSockets/WebSocketClient.cs:59]: This coupled with leaving some listeners fixes the issue. The rest is more unit test cruft.

File: src/MorseL.Scaleout.Redis/RedisBackplane.cs [hunk 2]
```diff
@@ -231,6 +231,7 @@ public RedisBackplane(IOptions<ConfigurationOptions> options)
             return RedisKeyGroupPrefix + group;
         }
 
+        internal IDictionary<string, Object> Connections => _connections;
         internal IDictionary<string, IDictionary<string, object>> Groups => _groups;
```
Comment (ryanwersal) [inline src/MorseL.Scaleout.Redis/RedisBackplane.cs:235]: Is it possible to type the objects as something more concrete? I haven't looked directly at usage yet but using object this pervasively seems less than ideal.

File: test/MorseL.Scaleout.Redis.Tests/EndToEndTests.cs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+namespace MorseL.Scaleout.Redis.Tests
+{
+    public class EndToEndTests
+    {
+        [Fact]
+        public async void ConnectionSubscriptionAddedOnConnect()
+        {
+            RedisBackplane backplane = null;
+            using (new SimpleMorseLServer<TestHub>(IPAddress.Any, 5000, (collection, builder) =>
```
Comment (ryanwersal) [inline test/MorseL.Scaleout.Redis.Tests/EndToEndTests.cs:19]: We should probably look at doing something like the builder pattern to reduce the setup noise in these tests (please, please don't just move it into a setup function 😿 )

File: src/MorseL.Client.WebSockets/WebSocketClient.cs (hunk 1 above)
Comment (ryanwersal) [inline src/MorseL.Client.WebSockets/WebSocketClient.cs:59]: Ha well that ended up being pretty minimal. Glad it wasn't too intense.

**Review by ryanwersal — APPROVED** [review]
LGTM

File: src/MorseL.Scaleout.Redis/RedisBackplane.cs (hunk 2 above)
Comment (berdon) [reply src/MorseL.Scaleout.Redis/RedisBackplane.cs:235]: This is me bastardizing ConcurrentDictionary for a concurrent hash. The values are always null.

Comment (ryanwersal) [reply src/MorseL.Scaleout.Redis/RedisBackplane.cs:235]: Ah 👍

**Pull request merged**